File: Python/21-30/e_30.py
# Escribe en python un programa que permita calcular el tiempo de ejecución de un script. Tomamos como ejemplo el programa del ejercicio 24.


import time

# Inicio del tiempo de ejecución
inicio_tiempo = time.time()

# Script original
numero = 7
print(f"Tabla de multiplicar del {numero}:")
for i in range(1, 11):
    resultado = numero * i
    print(f"{numero} x {i} = {resultado}")

# Fin del tiempo de ejecución
fin_tiempo = time.time()

# Calcular la duración
duracion = fin_tiempo - inicio_tiempo
print(f"\nEl script tomó {duracion:.6f} segundos en ejecutarse.")



# En este ejemplo, se utiliza time.time() para obtener el tiempo actual antes y después de ejecutar el script.
# La diferencia entre estos tiempos te dará la duración de la ejecución del script. 
# La duración se imprime al final del script.
# Ten en cuenta que la precisión de la medición de tiempo puede depender del sistema operativo y otros factores.

File: JS/21-30/e_30.js
// Inicio del tiempo de ejecución
const inicioTiempo = new Date();

// Script original
const numero = 7;
console.log(`Tabla de multiplicar del ${numero}:`);
for (let i = 1; i <= 10; i++) {
    const resultado = numero * i;
    console.log(`${numero} x ${i} = ${resultado}`);
}

// Fin del tiempo de ejecución
const finTiempo = new Date();

// Calcular la duración en milisegundos
const duracion = finTiempo - inicioTiempo;

console.log(`\nEl script tomó ${duracion} milisegundos en ejecutarse.`);



// En JavaScript, puedes utilizar el objeto Date para medir el tiempo de ejecución de un script. 
// En este ejemplo, se utiliza new Date() para obtener la marca de tiempo antes y después de ejecutar el script.
// La diferencia entre estas marcas de tiempo te dará la duración de la ejecución del script en milisegundos.
// La duración se imprime al final del script.
// Ten en cuenta que la precisión de la medición de tiempo puede depender del entorno de ejecución del navegador o Node.js.